hogql_parser/parser.cpp

```diff
@@ -1373,7 +1373,24 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
     RETURN_NEW_AST_NODE("CompareOperation", "{s:N,s:N,s:N}", "left", left, "right", null_constant, "op", op);
   }
 
-  VISIT_UNSUPPORTED(ColumnExprTrim)
+  VISIT(ColumnExprTrim) {
+    const char* name;
+    if (ctx->LEADING()) {
+      name = "trimLeft";
+    } else if (ctx->TRAILING()) {
+      name = "trimRight";
+    } else if (ctx->BOTH()) {
+      name = "trim";
+    } else {
+      throw ParsingException("Unsupported value of rule ColumnExprTrim");
+    }
+    auto string_literal_terminal = ctx->STRING_LITERAL();
+    if (!string_literal_terminal) throw PyInternalException();
+    string text = unquote_string_terminal(string_literal_terminal);
+    PyObject* value = build_ast_node("Constant", "{s:s#}", "value", text.data(), text.size());
+    if (!value) throw PyInternalException();
+    RETURN_NEW_AST_NODE("Call", "{s:s,s:[NN]}", "name", name, "args", visitAsPyObject(ctx->columnExpr()), value);
```

Comment:
It's not _completely_ kosher to use `visitAsPyObject()` as an arg directly (implied by point 1 of exception handling in the guide), because this may throw, in which case we'd be left with `value` dangling forever as its refcount wouldn't be decremented. Which is a potential memory leak, though tiny. But we can avoid the cleanup problem altogether by simply putting `expr = visitAsPyObject(ctx->columnExpr());` right before `build_ast_node()`

Reply:
cool, thanks! right above `build_ast_node` or even higher? 🤔

Reply:
Right above that or `unquote_string_terminal` if that's better for readability. Just needs to be before the first `PyObject*`-returning call